Account for different ordering of files when traversing filesystem

gaudetja opened the pull request:
When using this tool to build release binaries, I found it was important to force a consistent order of files fed to the linker across machines.

gaudetja commented:
@mvandervoord any interest in these changes for build-determinism across different machines? If not I'll close this as I have moved away from Ceedling for the project it impacted.

mvandervoord commented:
Hi

Yes. I planned to pull this change in. I just need to cherry pick it since it shouldn't have changes to the gem file here. :)

gaudetja commented:
> Hi
> 
> Yes. I planned to pull this change in. I just need to cherry pick it since it shouldn't have changes to the gem file here. :)

@mvandervoord Thank you for the review! I've rebased the change onto your `master` branch, and removed the version string change.
